meetings/working-groups/roles/roles-2023-01-25.md

```diff
@@ -0,0 +1,108 @@
+# Roles WG 2023-01-25
+
+## Language for relationships
+
+We'll use "augments" for relationship to underlying type 
+(comparable to "inherits" for relationship to base type).  
+We'll use "inherits" for relationship to inherited roles 
+(comparable to "implements" for relationship to implemented interfaces).  
+
+```csharp
+struct U { }
+role X : U { }
+role Y : U, X, X1 { }
+```
+"Y has underlying type U"  
+"Y augments U"  
+"Y inherits X and X1"  
+"Derived role Y inherits members from inherited roles X and X1"  
+
+## Syntax for role underlying type list
+
+We should assume multiple inheritance for now. Will update syntax to allow multiple inherited roles.  
+In that syntax, the first type will be the underlying type.  
+`role R : I { } // First type is underlying type (not an implemented interface)`
```

Comment:
I do feel like this syntax is a bit unwieldy. It mostly works ok for class inheritance, because there are only two categories (inherit and implement), and in all contexts the length of one of those categories is statically known (0 or 1 inherits for reference types, 0 implements for interfaces). For roles, this is not true: 1 underlying type, M inherited roles, and N implemented interfaces. I think that's likely to introduce complexity in understanding the code, for both roslyn and the human reader. Might be worth taking a note of this issue as an open question for the eventual syntax debate.

Reply:
I agree with you, it's not the clearer syntax.
I've already proposed using `for` https://github.com/dotnet/csharplang/issues/5497#issuecomment-984900268
But with the addition of multiples `roles`, maybe `role Role for UnderlyingType : Interface1, Interface2 with Role1, Role2 {}`
Everything is well separated and reads well for the user and the compiler.

Reply:
Yup. That's definitely a concern. I have an open issue on syntax in speclet.
C# went with a flat list (rather than separators and keywords) in the past, but it's definitely getting harder for this scenario.
A possible mitigation is to lean on naming convention (`I` prefix for interfaces helps, maybe a similar naming convention for roles?).

Reply:
Personally, I'm OK with a flat list for `class`es because it can only have 1 other `class`, the rest are `interface`s.
But for Roles, even with a potential prefix (`R...`) or suffix (`...Role`), I still prefer to have some sort of separator.